sftp-source/src/main/java/org/springframework/cloud/stream/module/sftp/source/SftpSource.java

```diff
@@ -0,0 +1,133 @@
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.sftp.source;
+
+import java.util.Collections;
+import java.util.regex.Pattern;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.cloud.stream.module.MaxMessagesProperties;
+import org.springframework.cloud.stream.module.PeriodicTriggerConfiguration;
+import org.springframework.cloud.stream.module.file.FileConsumerProperties;
+import org.springframework.cloud.stream.module.sftp.SftpSessionFactoryConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.SourcePollingChannelAdapterSpec;
+import org.springframework.integration.dsl.core.Pollers;
+import org.springframework.integration.dsl.sftp.Sftp;
+import org.springframework.integration.dsl.sftp.SftpInboundChannelAdapterSpec;
+import org.springframework.integration.dsl.support.Consumer;
+import org.springframework.integration.file.splitter.FileSplitter;
+import org.springframework.integration.file.transformer.FileToByteArrayTransformer;
+import org.springframework.integration.scheduling.PollerMetadata;
+import org.springframework.integration.sftp.filters.SftpRegexPatternFileListFilter;
+import org.springframework.integration.sftp.filters.SftpSimplePatternFileListFilter;
+import org.springframework.integration.sftp.session.DefaultSftpSessionFactory;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.scheduling.Trigger;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author Gary Russell
+ */
+@EnableBinding(Source.class)
+@EnableConfigurationProperties({SftpSourceProperties.class,
+		FileConsumerProperties.class, MaxMessagesProperties.class})
+@Import({PeriodicTriggerConfiguration.class, SftpSessionFactoryConfiguration.class})
+public class SftpSource {
+
+	@Autowired
+	private SftpSourceProperties properties;
+
+	@Autowired
+	private FileConsumerProperties fileConsumerProperties;
+
+	@Autowired
+	MaxMessagesProperties maxMessagesProperties;
+
+	@Autowired
+	DefaultSftpSessionFactory sftpSessionFactory;
+
+	@Autowired
+	Trigger trigger;
+
+	@Autowired
+	@Bindings(SftpSource.class)
+	Source source;
+
+	@Bean
+	public PollerMetadata poller() {
+		return Pollers.trigger(this.trigger).maxMessagesPerPoll(this.maxMessagesProperties.getMaxMessages()).get();
+	}
+
+	@Bean
+	public IntegrationFlow sftpInboundFlow() {
```

Comment:
I'm not familiar enough with SI DSL. What makes this context actually work? Are `IntegrationFlow` beans handled in some way? I don't see any SI annotation here.